### ISSUE TITLE
getting this warning while building apk

### ISSUE THREAD

**Issue by mnjammy007** [AppsFlyerSDK/appsflyer-flutter-plugin#280]
Note: ~\Pub\Cache\hosted\pub.dev\appsflyer_sdk-6.11.3\android\src\main\java\com\appsflyer\appsflyersdk\AppsflyerSdkPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: ~\Pub\Cache\hosted\pub.dev\appsflyer_sdk-6.11.3\android\src\main\java\com\appsflyer\appsflyersdk\AppsflyerSdkPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

**Comment by samdeane** [comment]
Still present with 6.14.3:

```
Note: /Users/sam/.pub-cache/hosted/pub.dev/appsflyer_sdk-6.14.3/android/src/main/java/com/appsflyer/appsflyersdk/AppsflyerSdkPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/sam/.pub-cache/hosted/pub.dev/appsflyer_sdk-6.14.3/android/src/main/java/com/appsflyer/appsflyersdk/AppsflyerSdkPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

**Comment by jannisnikoy** [comment]
And still present in 6.15.2 😅 Doesn't seem to cause any issues other than throwing a warning when building. Also not related to apk specifically, getting the same when building .aab.

**Comment by DanBeltranuwu** [comment]
And still present in 6.16.21

**Comment by roy-carlet** [comment]
And still present in 6.17.5

**Comment by Dani-Koza-AF** [comment]
Thank you for bringing this to our attention. I completely understand your concern, and I'd like to provide some clarity about this warning message.

## What's Happening

The warning you're seeing:
```
Note: .../appsflyer_sdk-6.17.5/android/src/main/java/.../AppsflyerSdkPlugin.java uses or overrides a deprecated API.
```

This is a **compile-time warning** from the Java compiler, not an error. It's informing you that the plugin contains code that uses or provides deprecated methods.

## Why This is Perfectly Normal (and OK!)

This is actually a **very common and expected practice** in plugin development, and here's why:

1. **Backward Compatibility**: We maintain deprecated methods to ensure that apps using older versions of our API continue to work without breaking. This is a standard practice in professional software development and demonstrates our commitment to not breaking existing integrations.

2. **Graceful Migration Path**: When we introduce improved APIs (like `setConsentDataV2` replacing the older `setConsentData`), we don't immediately remove the old method. Instead, we:
   - Mark it as `@Deprecated` to signal to developers that a better alternative exists
   - Keep it functional so existing code continues to work
   - Document the new recommended approach

3. **Flutter Ecosystem Standard**: This is how most well-maintained Flutter plugins handle API evolution. You'll find similar warnings in many popular plugins.

## Is This a Security or Functionality Issue?

**No!** The warning doesn't indicate:
- ❌ A security vulnerability
- ❌ Broken functionality
- ❌ Performance issues
- ❌ Code that will stop working

It simply means we're following best practices by providing a transition period for developers to migrate to newer APIs.

## What Should You Do?

**For most users: Nothing!** This warning can be safely ignored. Your app will compile and run perfectly fine.

If you want to suppress the warning in your build output, you can add this to your `android/app/build.gradle`:

```gradle
android {
    lintOptions {
        disable 'Deprecation'
    }
}
```

However, this is completely optional and cosmetic.

## Our Commitment

We continuously update the plugin to support the latest Flutter and native SDK features while ensuring existing integrations remain stable. Deprecation warnings are part of this commitment to both innovation and reliability.

If you have any questions or concerns about specific APIs or migration paths, please feel free to address to support team for  help.
Contact AppsFlyer support through the Customer Assistant Chatbot for assistance with troubleshooting issues or product guidance. To do so, please follow [this article](https://support.appsflyer.com/hc/en-us/articles/23583984402193-Using-the-Customer-Assistant-Chatbot). 


---

**TL;DR**: This is a normal, harmless compile-time warning that indicates good software engineering practices. Your app is safe and will work perfectly! 🎉
